kt/common/src/main/kotlin/godot/common/extensions/StringExtensions.kt

```diff
@@ -48,6 +48,48 @@ fun String.convertToSnakeCase(): String =
         else accumulator.append(character)
     }.toString()
 
+fun String.toUpperSnakeCase(): String {
+    if (this.isEmpty()) return this
+
+    val result = StringBuilder()
+    for (i in this.indices) {
+        val currentChar = this[i]
+        if (i > 0 && currentChar.isUpperCase()) {
+            val prevChar = this[i - 1]
+            // Insert underscore if the previous character is lowercase
+            // OR if the previous character is uppercase but the next character exists and is lowercase
+            if (prevChar.isLowerCase() || (prevChar.isUpperCase() && i < this.lastIndex && this[i + 1].isLowerCase())) {
+                result.append('_')
+            }
+        }
+        result.append(currentChar)
+    }
+    return result.toString().uppercase(Locale.getDefault())
+}
+
+fun String.removePrefixWords(wordWithDashes: String): String {
+    // We split the 2 strings into different "words"
+    val otherWords = wordWithDashes.split('_')
+    val valueWords = this.split('_')
+
+    var index = 0
+    for (enumWord in otherWords) {
+        if (index >= valueWords.size) break
```

Comment:
just a `otherWords.forEach` would do the trick?

Reply:
You can't break in a foreach, if you write a return@ForEach, it's continue to be equivalent to a continue.

Reply:
didn't register you looped over `otherWords` and checked against `valueWords` sorry about that